laser_topic: "/scan"
wheel_odom_topic: "/odom"
imu_topic: "/camera/imu"
camera_topic: "/camera/color/image_raw" # antything you want

enable_camera: false    #awaly false
enable_laser: true      
enable_camera_vis: true #awaly true
enable_laser_vis: true



T_imu_to_wheel: [ 0.174472  , -0.981392 ,  0.0801839 , -0.0727832  ,
                 -0.0500957 , -0.090174 , -0.994665  ,  0.38917205 ,
                  0.983387  , 0.169525  , -0.0648965 , -0.128696   ,
                  0.0       , 0.0       ,  0.0       ,  1.0 ]



#no estimate
#read from camera
T_imu_to_camera: [ 0.99054   ,   -0.04438   ,   0.129849     ,  -0.020  ,
                   0.0465    ,   0.998818   ,  -0.0139036    ,   0.005  ,
                  -0.129078  ,   0.019819   ,   0.991436     ,   0.012  ,
                   0.0       ,   0.0        ,   0.0          ,   1.0]



#no estimate
#estimated by lidar_wheel_calibration
T_imu_to_laser: [  0.144421  ,  -0.986262  ,   0.0801839 ,  0.000319325,
                  -0.0528261 ,  -0.0886025 ,  -0.994665  ,  0.104101,
                   0.988105  ,   0.139415  ,  -0.0648965 ,  0.0228106,
                   0.0       ,   0.0       ,   0.0       ,  1.0 ]


camera_K: [ 455.66 , 0.0     , 324.146 ,
            0.0    , 455.086 , 181.557 ,
            0.0    , 0.0     , 1.0     ]


FPS: 40

g: 9.8


#noise
#each second
imu_noise_acc_sigma: [ 0.0163  ,0.0163  , 0.0163  ]
imu_bias_acc_sigma:  [ 0.00499 , 0.00499  , 0.00499 ]
imu_noise_gyro_sigma: [ 0.003208 , 0.003208   , 0.003208  ]
imu_bias_gyro_sigma:  [ 0.000499 , 0.000499  , 0.000499 ]

#per meter
# len dir yaw
wheel_sigma: [0.1 , 99999.0 , 999.99]



#each pixel
camera_sigma: [ 20.0, 20.0 ]
max_camera_reproject_error: 20
max_camera_feature_dis: 50.0


#manifold noise
manifold_p_sigma: 0.01    #m
#manifold_q_sigma: 0.00001    #rad
manifold_q_sigma: 0.001    #rad


#tracking
max_feature_num: 5
feature_min_dis: 50.0


#Initialization parameters
slide_window_size : 10
p_motion_threshold : 0.1
q_motion_threshold : 0.05


#laser parameters
w_laser_each_scan: 100.0
h_laser_each_scan: 100.0
laser_resolution: 0.05 

#meter
line_continuous_threshold: 0.1
line_min_len: 0.05
line_max_dis: 0.03

#degree
line_max_tolerance_angle: 175.0
line_to_line_sigma: 0.001


#keyfram param
key_frame_p_motion_threshold : 0.15
key_frame_q_motion_threshold : 0.15


#laser loop
a_res: 0.05  
d_res: 0.05
submap_count: 10
laser_loop_min_match_threshold: 8
loop_detect_min_interval: 100
verify_loop_rate: 0.5
loop_max_dis: 999.0
loop_edge_k: 10.0
loop_max_tf_p: 999.0
loop_max_tf_q: 999.0

loop_sigma_p: [0.01, 0.01, 0.01]
loop_sigma_q: [0.001, 0.001, 0.001]

#backend
use_ground_p_factor: true
use_ground_q_factor: false



#front end
ref_motion_filter_p: 0.01
ref_motion_filter_q: 0.01
ref_n_accumulation: 2
fast_mode: false
sliding_window_size: 5

#ouput
output_tum: false
output_dir: "/dev/null/"